app/services/MatcherPool.scala

```diff
@@ -167,23 +167,26 @@ class MatcherPool(val maxCurrentJobs: Int = 8, val maxQueuedJobs: Int = 1000, va
   }
 
   private def runValidationJob(job: MatcherJob): Future[(MatcherJob, List[RuleMatch])] = {
-    val jobResults = job.categoryIds.map { categoryId =>
-      matchers.get(categoryId) match {
-        case Some((_, matcher)) =>
-          val eventuallyMatches = matcher.check(MatcherRequest(job.blocks, categoryId))
-          job.promise.completeWith(eventuallyMatches)
```

Comment:
This is the problem – we're mapping across a list of many category ids, but a job can only be completed once.

So the job is completed with the first category, and returns a result; meanwhile other categories are checked, and their results are discarded.

Reply:
Thanks for the explanation, that does make sense 👍